plugins/ag-charts-generate-example-files/src/executors/generate/generator/utils/getDarkModeSnippet.ts

```diff
@@ -2,18 +2,28 @@ export const DARK_MODE_START = '/** DARK MODE START **/';
 export const DARK_MODE_END = '/** DARK MODE END **/';
 
 const DARK_MODE_SNIPPETS = {
-    vanilla: () => `
-        options.theme = {
-            ...options.theme,
-            baseTheme: localStorage['documentation:darkmode'] === 'true' ? 'ag-default-dark' : 'ag-default'
-        }
-        agCharts.AgCharts.update(chart, options);
-        document.querySelector('html').setAttribute('data-dark-mode', localStorage['documentation:darkmode'] === 'true');
-        window.addEventListener('message', (event) => {
-            if (event.data?.type === 'color-scheme-change') {
-              options.theme.baseTheme = event.data.darkmode ? 'ag-default-dark' : 'ag-default';
-              agCharts.AgCharts.update(chart, options);
-              document.querySelector('html').setAttribute('data-dark-mode', event.data.darkmode);
+    global: () => `
+        const applyDarkmode = (darkmode) => {
```

Comment:
This can be done for all frameworks (ignoring the actual framework code), but that's a problem for another day